src/common/src/consistent_hash/vnode.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// `VirtualNode` is the logical key for consistent hash. Virtual nodes stand for the intermediate
+/// layer between data and physical nodes.
+pub type VirtualNode = u16;
+pub const VNODE_BITS: usize = 11;
+pub const VIRTUAL_NODE_COUNT: usize = 1 << VNODE_BITS;
+pub const VNODE_BITMAP_LEN: usize = 1 << (VNODE_BITS - 3);
+
+// `VNodeBitmap` is a bitmap of vnodes for a state table, which indicates the vnodes that the state
+// table owns.
+#[derive(Clone, Default)]
+pub struct VNodeBitmap {
+    table_id: u32,
```

Comment:
Looks a little strange. Are you sure every `VNodeBitmap` needs a `table_id`?

Reply:
There will be a `VNodeBitmap` in every `Keyspace` of a state table, so `table_id` is in need here. Also, `VNodeBitmap` will be added to the interface of `StateStore`, where we use it for read pruning. In this case, `table_id` is also necessary. Any other case to use `VNodeBitmap`?

Reply:
And from the aspect of logic, the struct is designed for read pruning, which depends on state table. So having a `table_id` is reasonable, right? 🤔

Reply:
Sounds reasonable 😁